libs/components/src/shared/design-system/index.ts

```diff
@@ -1,12 +1,13 @@
 import { DesignSystem } from '@microsoft/fast-foundation';
 
-
-export const getPrefix = (url:string) => new URL(url).searchParams.get('prefix') || 'vwc';
+const defaultPrefix = 'vwc';
 
 // common components dependency
 /**
  * @param {HTMLElement} element - The element to get or create a design system for.
  * @param prefix
  * @returns {DesignSystem} - Represents a configurable design system. An API gateway to design system features.
  */
-export const designSystem = DesignSystem.getOrCreate();
+const designSystem = DesignSystem.getOrCreate();
+
+export const registerFactorial = (...args: any[]) => (prefix = defaultPrefix) => designSystem.withPrefix(prefix).register(...args);
```

Comment:
Unless I'm missing something, there is no need to destructure your arrays (creating copies) every time?
```suggestion
export const registerFactorial = (args: any[]) => (prefix = defaultPrefix) => designSystem.withPrefix(prefix).register(args);
```
and in every definition file, just call `export const registerSomething = registerFactorial(somethingElements);`

Reply:
It accepts parameters not array. Did u try?

Reply:
oops, you're right, suggestion updated.

Reply:
The new suggestion will assign only the 1st argument to `args`

Reply:
in definition.ts files:
```js
export const widgetElements = [widget, ...iconElements, ...focusElements];
```
`widgetElements` is an **array**, that you pass to `registerFactorial`:
```js
export const registerWidget = registerFactorial(widgetElements);
```
thus, in the shared util file:
```js
export const registerFactorial = (args: any[]) => (prefix = defaultPrefix) => designSystem.withPrefix(prefix).register(...args);
```
`registerFactorial` is defined as a function accepting an **array**, which is later spread since register accepts 1..n arguments.

your original code defines `registerFactorial` as a function with 1..n arguments (using the spread operator), which forces you to spread each `widgetElements` when calling it instead of simply passing its reference as I suggest.

Reply:
thanks! commited